### PR TITLE
adding pull approve

### PR DESCRIPTION
**Pull request by lomudi**
## Description

Adding pull approve github app.

## Checklist

* [x] PR is scoped to one task
* [ ] Tests are included
* [ ] Documentation is included

* [x] This PR is not making any UI change
* [ ] This PR is making a UI change
  * [ ] Screenshot(s) included

## QA :

* required test cases:
* specific apps / plugins to test :

### CODE REVIEW

File: .pullapprove.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+version: 3
```
Comment (lomudi) [inline .pullapprove.yml]: /pullapprove-test

```
version: 3

pullapprove_conditions:
- "'WIP' not in title"
- "'ci/circleci: build' in statuses.succeeded"

notifications:
- when: pull_request.opened
  if: "author_association in ['FIRST_TIMER', 'FIRST_TIME_CONTRIBUTOR', 'CONTRIBUTOR']"
  comment: >
    Hey @{{ author }}, please make sure your tests pass and you have looked everything over. Then we will review.

    Thanks for the PR!
- when: pullapprove.approved
  comment: "The review is completed. @{{ author }}, please merge this in when ready."

groups:
  code:
    reviewers:
      teams:
      - applicaster/dev-rel
    reviews:
      required: 1  
      request: 1  
      request_order: random  
```

**Comment by lomudi** [comment]
/pullapprove-test

```
version: 3

pullapprove_conditions:
- "'WIP' not in title"
- "'ci/circleci: build' in statuses.succeeded"

notifications:
- when: pull_request.opened
  if: "author_association in ['FIRST_TIMER', 'FIRST_TIME_CONTRIBUTOR', 'CONTRIBUTOR']"
  comment: >
    Hey @{{ author }}, please make sure your tests pass and you have looked everything over. Then we will review.

    Thanks for the PR!
- when: pullapprove.approved
  comment: "The review is completed. @{{ author }}, please merge this in when ready."

groups:
  code:
    reviewers:
      teams:
      - applicaster/dev-rel
    reviews:
      required: 1  
      request: 1  
      request_order: random  
```

File: .pullapprove.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (lomudi) [inline .pullapprove.yml]: /pullapprove-test

```
version: 3

pullapprove_conditions:
- "'WIP' not in title"
- "'ci/circleci: build' in statuses.succeeded"

notifications:
- when: pull_request.opened
  if: "author_association in ['FIRST_TIMER', 'FIRST_TIME_CONTRIBUTOR', 'CONTRIBUTOR']"
  comment: >
    Hey @{{ author }}, please make sure your tests pass and you have looked everything over. Then we will review.

    Thanks for the PR!
- when: pullapprove.approved
  comment: "The review is completed. @{{ author }}, please merge this in when ready."

groups:
  code:
    reviewers:
      users:
      - lomudi
      - liviur
      - snooky23
      teams:
      - applicaster/dev-rel
    reviews:
      required: 1  
      request: 1  
      request_order: random  
```

File: .pullapprove.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+pullapprove_conditions:
+- "'WIP' not in title"
```
Comment (snooky23) [inline .pullapprove.yml]: what about adding:
- "base.ref == 'master'"  # only review things being merged into

File: .pullapprove.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- "'ci/circleci: build' in statuses.succeeded"
+
+notifications:
+- when: pull_request.opened
+  if: "author_association in ['FIRST_TIMER', 'FIRST_TIME_CONTRIBUTOR', 'CONTRIBUTOR']"
+  comment: >
```
Comment (snooky23) [inline .pullapprove.yml]: do you sure we need the `>` sign?

File: .pullapprove.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    Thanks for the PR!
+- when: pullapprove.approved
+  comment: "The review is completed. @{{ author }}, please merge this in when ready."
+
+groups:
+  code:
+    reviewers:
+      users:
+      - lomudi
```
Comment (snooky23) [inline .pullapprove.yml]: If the team has a group named `applicaster/dev-rel` there is no need to set those users.

File: .pullapprove.yml (hunk 3 above)
Comment (lomudi) [reply .pullapprove.yml]: done

File: .pullapprove.yml (hunk 5 above)
Comment (lomudi) [reply .pullapprove.yml]: done

File: .pullapprove.yml (hunk 4 above)
Comment (lomudi) [reply .pullapprove.yml]: I can see it in their examples... https://docs.pullapprove.com/

**Pull request merged**